intake_erddap/erddap_cat.py

```diff
@@ -26,20 +33,58 @@ class ERDDAPCatalog(Catalog):
     def __init__(
         self,
         server: str,
-        kwargs_search: Optional[dict] = None,
+        kwargs_search: Optional[Dict[str, Union[str, int, float]]] = None,
+        category_search: Optional[List[str]] = None,
         erddap_client: Optional[Type[ERDDAP]] = None,
-        **kwargs
+        **kwargs,
     ):
         """ERDDAPCatalog initialization
 
-        Arguments
-        ---------
-
-            server
+        Parameters
+        ----------
+        server : str
+            ERDDAP server address, for example: "http://erddap.sensors.ioos.us/erddap"
+        kwargs_search : dict, optional
+            Keyword arguments to input to search on the server before making the catalog. Options are:
+            * to search by bounding box: include all of min_lon, max_lon, min_lat, max_lat: (int, float)
+              Longitudes must be between -180 to +180.
+            * to search within a datetime range: include both of min_time, max_time: interpretable
+              datetime string, e.g., "2021-1-1"
+        category_search : list, optional
+            Use this to narrow search by ERDDAP category. The syntax is `[category, key]`, e.g.
+            ["standard_name": "temp"]. `category` is the ERDDAP category for filtering results. Good
```

Comment:
I recommend changing the type to a tuple because it's a fixed length input. Also, I think that the `:` in the example should be a `,`.